client/state/data-layer/wpcom/sites/simple-payments/index.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Formats /posts endpoint response into a product list
+ * @param { Object } product raw /post endpoint response to format
+ * @returns { Object } sanitized and formatted product
+ */
+function customPostToProduct( product ) {
+	return Object.assign(
+		{
+			ID: product.ID,
+			description: product.content,
+			title: product.title
+		},
+		product.metadata.reduce( reduceMetadata, {} )
+	);
+}
+
+/**
+ * Transforms a product definition object into proper custom post type
+ * @param { Object } product action with product payload
+ * @returns { Object } custom post type data
+ */
+function productToCustomPost( product ) {
+	return Object.keys( product ).reduce(
+		function( payload, current ) {
+			if ( metadataSchema[ current ] ) {
+				payload.metadata.push( {
+					key: metadataSchema[ current ].metaKey,
+					value: product[ current ]
+				} );
+			}
+			return payload;
+		},
+		{
+			type: 'jp_pay_product',
+			metadata: [],
+			title: product.title,
+			content: product.description,
+		}
+	);
+}
+
+/**
+ * Issues an API request to fetch media for a site and query.
+ *
+ * @param  {Object}  store  Redux store
+ * @param  {Object}  action Action object
+ * @return {Promise}        Promise
+ */
+export function requestSimplePaymentsProducts( { dispatch, getState }, { siteId } ) {
+	if ( isRequestingSimplePaymentsProductList( getState(), siteId ) ) {
+		return;
+	}
+
+	dispatch( requestingProductList( siteId ) );
+	log( 'Request product list for site %d.', siteId );
+
+	return wpcom
+		.site( siteId )
+		.postsList( {
+			type: 'jp_pay_product',
+			status: 'publish'
+		} )
+		.then( ( { found, posts } ) => {
+			dispatch( receiveProductsList( siteId, found, posts.map( customPostToProduct ) ) );
+			dispatch( successProductListRequest( siteId ) );
+		}	)
+		.catch( err => dispatch( failProductListRequest( siteId, err ) ) );
+}
+
+/**
+ * Issues an API request to add a new product
+ * @param {Object} store Redux store
+ * @param {Object} action Action object
+ */
+export function requestSimplePaymentsProductAdd( { dispatch }, action ) {
+	const { siteId, product } = action;
+
+	dispatch( http( {
+		method: 'POST',
+		path: `/sites/${ siteId }/posts/new`,
+		body: productToCustomPost( product ),
+	}, action ) );
+}
+
+export const receiveUpdateProduct = ( { dispatch }, action, next, newProduct ) => {
+	const { siteId } = action;
+
+	dispatch( {
+		siteId,
+		product: customPostToProduct( newProduct ),
+		type: SIMPLE_PAYMENTS_PRODUCTS_LIST_RECEIVE_UPDATE,
+	} );
+};
+
+/**
+ * Issues an API request to edit a product
+ * @param {Object} store Redux store
+ * @param {Object} action Action object
+ * @return {Promise} Promise
+ */
+export function requestSimplePaymentsProductEdit( { dispatch }, action ) {
+	return wpcom
+		.site( action.siteId )
+		.post( action.productId )
+		.update( productToCustomPost( action.product ) )
+		.then( ( newProduct ) => {
+			dispatch( receiveUpdateProduct( action.siteId, customPostToProduct( newProduct ) ) );
```

Comment:
I know this is in-progress, but be careful which version of `receiveUpdateProduct` you're using. 😁